tempesta_fw/http_msg.c

```diff
@@ -1043,3 +1047,146 @@ __tfw_http_msg_alloc(int type, bool full)
 	return hm;
 }
 
+/**
+ * Determination length of the header's real part (for details see comment
+ * for @tfw_http_msg_hdr_write() below) to store it in the encoder dynamic
+ * index.
+ */
+unsigned long
+tfw_http_msg_hdr_length(const TfwStr *hdr, unsigned long *name_len,
+			unsigned long *val_off, unsigned long *val_len)
```

Comment:
I believe the function should be much simpler. With the recent changes in `RGEN_HDR_OTHER()` all the headers store name and values separately. The only needed change is to add `TFW_STR_NAME` and `TFW_STR_VALUE` flags setting to `RGEN_HDR_OTHER()`. Next, we probably can store field value right after the spaces, but I'm not sure - need to check serialization and security logic. So the function can just iterate the chunks and sum their lengths. In worst case it also can skip spaces in first value chunks.

Reply:
It seems that in general case we do not set `TFW_STR_NAME`/`TFW_STR_VALUE` flags for parse headers (`__TFW_HTTP_PARSE_SPECHDR_VAL` and `__TFW_HTTP_PARSE_RAWHDR_VAL` macros). But the more difficult issue here is that in general case the end OWS are placed into the value chunk during headers parsing (e.g. `Req_I_EoT` state in the `__req_parse_accept()`, `__req_parse_cache_control()` etc.).

Reply:
Discussion in chat results in necessity of appropriate changes in HTTP parser to allow faster determination of header's name and value parts in `TfwStr` chunks. I'll made that changes in context of other corrections in HTTP parser in next/last PR.